### DISCUSSION TITLE
想了解一下绕过国内IP部分是怎么样的流程。

### DISCUSSION THREAD

**Discussion by Vegas7** [nikkinikki-org/OpenWrt-nikki#217] [Q&A] [answered]
是否像Openclash一样，匹配国内域名表，然后通过预设DNS来解析到IP，命中国内IP则不走内核？
如果是的话，那么这个预设DNS解析到的结果IP，是作为最终连接的IP，还是会将域名重新交给上游DNS？
（担心的点是如果没有走上游运营商DNS，得不到最佳CDN）

**Comment by morytyann** [comment, marked as answer]
不是，是直接匹配IP，而且现在OpenClash也是这么做的了

**Reply by Vegas7** [reply]
> 不是，是直接匹配IP，而且现在OpenClash也是这么做的了

那如果先经过上游解析了，那不是全泄露了一遍。

**Reply by morytyann** [reply]
我都说了不是OC的逻辑，没有什么预设DNS。

**Reply by Vegas7** [reply]
> 我都说了不是OC的逻辑，没有什么预设DNS。

抱歉不是很清楚具体逻辑。
能否简单说明下：

1. 如果要在经过内核之前解析获得IP进行判断，那么这个阶段是如何实现的？
2. 如果解析后判断需要进入内核，那么是如何重新将域名返回内核的，是类似mapping机制？

**Reply by morytyann** [reply]
插件只做了一件事，就是绕过目标IP是中国大陆IP的连接，DNS解析是Mihomo做的。

**Reply by Vegas7** [reply]
> 插件只做了一件事，就是绕过目标IP是中国大陆IP的连接，DNS解析是Mihomo做的。

不是很理解，如果在不进入内核的情况下，域名需要解析成IP来做判断是否为大陆，那么这次解析也是Mihomo做的？

**Reply by morytyann** [reply]
非要回答的话

1. 没有任何实现，Mihomo解析的
2. 由于是Mihomo解析的，Mihomo自然有对于的映射关系

就这么简单

**Reply by morytyann** [reply]
可算明白了？所以说不要套OC的逻辑，我猜你想问Fake-IP怎么办，答案是`fake-ip-filter`现在支持geosite了。

**Reply by Vegas7** [reply]
> 非要回答的话
> 
> 1. 没有任何实现，Mihomo解析的
> 2. 由于是Mihomo解析的，Mihomo自然有对于的映射关系
> 
> 就这么简单

了解了，谢谢。
